src/main/kotlin/uk/gov/justice/digital/hmpps/courtlistsplitter/model/externaldocumentrequest/DocumentInfoDeserializer.kt

```diff
@@ -0,0 +1,58 @@
+package uk.gov.justice.digital.hmpps.courtlistsplitter.model.externaldocumentrequest
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.core.TreeNode
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.courtlistsplitter.model.externaldocumentrequest.OuCodeDeserializer.Companion.OU_CODE_LENGTH
+import java.io.IOException
+import java.time.LocalDate
+import java.time.Month
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeParseException
+
+@Component
+class DocumentInfoDeserializer<T>(clazz: Class<Info>) : StdDeserializer<Info>(clazz) {
+
+  @JsonCreator
+  constructor() : this(Info::class.java)
+
+  @Throws(IOException::class)
+  override fun deserialize(jp: JsonParser, context: DeserializationContext): Info {
+    val jsonNode: JsonNode = jp.codec.readTree<TreeNode>(jp).get(Info.SOURCE_FILE_NAME_ELEMENT) as JsonNode
+    val sourceFileName: String = jsonNode.asText("")
+
+    // Source filename has the following format 146_27072020_2578_B01OB00_ADULT_COURT_LIST_DAILY
+    val fileNameParts = sourceFileName.split(FIELD_DELIM.toRegex()).toTypedArray()
+    if (fileNameParts.size < 4) {
+      log.error("Unable to determine OU code and date of hearing from source file name of {}", sourceFileName)
+      Info(-1, "", LocalDate.of(1970, Month.JANUARY, 1))
```

Comment:
Should there not be return statements before these `Info` constructors? Is this a Kotlin thing I'm about to learn about? 😄

Reply:
I'm going to try and answer my own question - Presumably Kotlin can infer that this in fact the return value because as it's neither operated on or assigned to a variable it would otherwise be redundant?

Reply:
The above only applies if the creation of the Info has no side effects of course...

Reply:
I was going to consult with you aout `Info` in any case. So it may appear more than once in any message. The only thing of interest in the `source_file_name` which we parse to get date of hearing, sequence number and a court code. The date of hearing and sequence values were previously useful when we had that period of duplicate messages. 

The court code was useful when the `ou_code` value wasn't appearing in the `session` itself. But CGI added that some time back now. So here's the thing - the `ou_code` seems now to be in the session for each message we receive. We still have code which falls back to the Info for court code if it isn't. Maybe the time has come to remove Info entirely but I haven't checked that `ou_code` really is in every session. In the XSD (which I know isn't actually enforced), it is optional. But then so is `source_file_name` and that has always been present.

I don't like setting arbitrary values but it seems better than leaving them null and dealing with that everywhere else. 

The other thing is that in this case, because CPG checks and parses `source_info_file` in each and every message for filtering, if Info isn't right then we won't see it in the splitter. 

As an action, what about monitoring the ou_code in session ?

Reply:
Added code to log when we go to Info for court code